lib/dialects/abstract/connection-manager.js

```diff
@@ -295,6 +295,11 @@ class ConnectionManager {
   releaseConnection(connection) {
     return this.pool.release(connection).tap(() => {
       debug('connection released');
+    }).catch((e) => {
+      // ISSUE(7625) No-op if generic-pool is unable to release the connection, it may have already been released.
+      if (!/Resource not currently part of this pool/.exec(e.message)) {
```

Comment:
I am just giving support for @jaredjenkins description where he says "generic-pool does allow suggest that you can no-op on these errors if they aren't a major issue. It's bit sketch, but we could no-op when we see the "Resource not currently part of this pool" err and throw otherwise."